target/snitch_cluster/Makefile

```
@@ -218,12 +232,30 @@ test/bootrom.elf test/bootrom.dump test/bootrom.bin: test/bootrom.S test/bootrom
 	riscv64-unknown-elf-objcopy -j .text -O binary test/bootrom.elf test/bootrom.bin
 
 
-$(SNAX_GEN):
-	${WRAPPER_GEN} --cfg_path="$(ROOT)/target/snitch_cluster/$(CFG_OVERRIDE)" \
+SNAX_TEMPLATE_SOURCES = $(wildcard $(SNAX_TPL_PATH)*.tpl)
+CFG_FILE = $(ROOT)/target/snitch_cluster/$(CFG_OVERRIDE)
+SNAX_SCALA_SOURCES = \
+	$(wildcard ${CHISEL_PATH}/src/main/scala/snax/streamer/*.scala) \
+	$(wildcard ${CHISEL_PATH}/src/main/scala/snax/csr_manager/*.scala)
+
+# Pick the first file
+FIRST_SNAX_GEN := $(firstword $(SNAX_GEN))
+# Pick all the files but the first file, if there are more than one
+OTHER_SNAX_GEN := $(filter-out $(FIRST_SNAX),$(SNAX_GEN))
+
+# Force a serialized build in parallel make through a circular dependency
+# Implemented as documented in:
+# https://www.gnu.org/software/automake/manual/html_node/Multiple-Outputs.html
+# Please note the limitations of this method over there
+$(FIRST_SNAX_GEN) : $(SNAX_TEMPLATE_SOURCES) $(CFG_FILE) $(SNAX_SCALA_SOURCES)
```

Would it still work even though it's:

```suggestion
$(FIRST_SNAX_GEN) : $(CFG_FILE) 
```

---

That would work, but then make will not build anymore if SNAX_TEMPLATE_SOURCES or SNAX_SCALA_SOURCES are altered after FIRST_SNAX_GEN is built. Make can also not throw an error anymore if  any of the dependencies are missing.